subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/NormalizedPathFingerprintCompareStrategy.java

```diff
@@ -63,6 +64,16 @@ protected boolean doVisitChangesSince(
         Map<String, FileSystemLocationFingerprint> previousFingerprints,
         String propertyTitle,
         boolean shouldIncludeAdded
+    ) {
+        Preconditions.checkArgument(shouldIncludeAdded);
```

Comment:
If you add this check then please update the Javadoc too. I'm not sure why separating the two methods is useful—as opposed to simply adding this line in the existing method. Are we planning to drop this method at some point in favor of the new one?

Reply:
@wolfs mentioned `shouldIncludeAdded` is always true here.
Please tell me how to proceed.

Reply:
Updated the doc. If you want to change it further, please push the change here, it's simpler that way.
I've extracted the method since we found it to be cleaner than asserting the value and making it accessible afterwards.

Reply:
@lptr Only `AbsolutePathFingerprintCompareStrategy` is used for the outputs, so only this one needs to support `includeAdded`. Therefore, we should only support `includeAdded=true` here, since the strategy is only used for inputs.

It would probably be better to introduce a new type (maybe FingerprintCompareStrategy vs. OutputFingerprintCompareStrategy?) WDYT?

Reply:
I agree that it would be best to refactor, but let's do it in another PR. I'd leave everything here as it was and open an issue describing how the API around `includeAdded` is bad with any suggestions on how to refactor it. We should quickly discuss the plans and then we can fix it. Does that sound good to you two?

Reply:
Yes, and if this PR doesn't make it worse, can we merge it before deciding?